### PR TITLE
Layout component-section container content

### PR DESCRIPTION
**Pull request by mahidhar-reddy09**
Set up layout component for section,container and content.

### CODE REVIEW

File: src/layout/container.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+export const Container = React.forwardRef(Component);
+
+interface StyleProps {
+    $type: ContainerType;
+    $stretch: boolean;
+}
+
+const StyledContainer = styled.div<StyleProps>`
```
Comment (qroll) [inline src/layout/container.tsx:35]: could you help to move the styled component to a separate file?

File: src/layout/types.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export interface CommonLayoutProps
+    extends React.HTMLAttributes<HTMLDivElement> {
+    children: React.ReactNode;
+    "data-testid"?: string | undefined;
+    stretch?: boolean;
```
Comment (qroll) [inline src/layout/types.ts:5]:
```suggestion
    stretch?: boolean | undefined;
```

File: tests/layout/layout-container.spec.tsx
Comment (qroll) [inline tests/layout/layout-container.spec.tsx:1]: unit tests are not required for these components as they have minimal behaviour and the logic is mostly for visual styling

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/layout/container.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+    ${MediaQuery.MaxWidth.sm} {
+        width: 100%;
+        padding: 0;
+        max-width: unset;
+    }
+
+    ${(props) => {
+        switch (props.$type) {
+            case "grid":
```
[the remark this answers is not in the document]
Comment (mahidhar-reddy09) [reply src/layout/container.tsx:70]: Hi Roo Ling I am not sure what does comment means. In the sense that what or how is the theme and breakpoint used other than the current mediaquery breakpoint in the context of this grid case?

File: src/layout/section.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            `;
+        } else {
+            return css`
+                padding: 0 1.5rem;
+
+                padding-left: 1.5rem !important;
+                padding-right: 1.5rem !important;
+            `;
+        }
+    }}
```
Comment (qroll) [inline src/layout/section.tsx:54]: since the margin will be in the container, this doesn't seem to be in use any more

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/layout/container.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        padding: 0 ${Breakpoint["sm-margin"]};
+    }
+
+    ${MediaQuery.MaxWidth.xs} {
+        padding: 0 ${Breakpoint["xs-margin"]};
+    }
+
+    ${MediaQuery.MaxWidth.xxs} {
+        padding: 0 ${Breakpoint["xxs-margin"]};
+    }
```
Comment (qroll) [inline src/layout/container.styles.tsx:53]: this would take override the styles above. only apply this if `$stretch` is false.

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/layout/container.styles.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    height: auto;
+
+    ${(props) => {
+        if (props.$stretch) {
+            return css`
+                padding: 0 32px;
+            `;
+        } else {
+            return css`
+                padding: 0 32px;
```
Comment (qroll) [inline src/layout/container.styles.tsx:25]: should be based on xxl-margin token

**Review by qroll — APPROVED** [review]

**Pull request merged**